pkg/providers/cloudstack/cloudstack.go

```diff
@@ -87,9 +91,56 @@ func (p *cloudstackProvider) PostWorkloadInit(ctx context.Context, cluster *type
 }
 
 func (p *cloudstackProvider) UpdateSecrets(ctx context.Context, cluster *types.Cluster) error {
+	contents, err := p.createSecrets(ctx, cluster)
+	if err != nil {
+		return fmt.Errorf("creating secrets object: %v", err)
+	}
+
+	err = p.providerKubectlClient.ApplyKubeSpecFromBytes(ctx, cluster, contents)
```

Comment:
could you add more validation here to make sure the secret does exist and that it aligns with the local environment's secret values?

Reply:
Added a step checking the secrets are aligned with the env vars in the upgrade cluster preflight checks. 
https://github.com/aws/eks-anywhere/pull/2666/commits/0e663e16eee6462009bb65b6b917e469b81c9a86